keel-core/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -220,21 +225,24 @@ class ApplicationService(
   }
 
   private fun buildArtifactSummaryInEnvironment(deliveryConfig: DeliveryConfig, environmentName: String, artifact: DeliveryArtifact, version: String, status: PromotionStatus): ArtifactSummaryInEnvironment? {
+    val artifactGitMetadata = getArtifactInstance(artifact, version)?.gitMetadata
+    val potentialSummary = getPotentialSummary(deliveryConfig, environmentName, artifact.reference, version)
+
     return when (status) {
-      PENDING -> ArtifactSummaryInEnvironment(
-        environment = environmentName,
-        version = version,
-        state = status.name.toLowerCase()
-      )
+      PENDING -> {
+        val newerGitMetadata = repository.getGitMetadataByPromotionStatus(deliveryConfig, environmentName,  artifact.reference, CURRENT.name)
+
+        ArtifactSummaryInEnvironment(
+          environment = environmentName,
+          version = version,
+          state = status.name.toLowerCase(),
+          // comparing CURRENT (new code) vs. PENDING (version is question, old code)
+          diffLink = generateDiffLink(newerGitMetadata, artifactGitMetadata)
+        )
+      }
       SKIPPED -> {
         // some environments contain relevant info for skipped artifacts, so
```

Comment:
This comment was about the call immediately below, which you moved to the top of the function. Please move the comment as well.

Reply:
done